src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -3545,6 +3546,29 @@ internal static class MemberInvocationLoggingOps
             }
         );
 
+        private static string ArgumentToString(object arg)
+        {
+            object baseObj = PSObject.Base(arg);
+            Type baseType = baseObj?.GetType();
+
+            switch (baseObj)
+            {
+                // The argument is null or AutomationNull.Value.
+                case null: return "null";
+
+                // The argument is a pure PSObject without a base object.
+                case PSObject: return "PSObject{}";
+
+                // The argument is not a PSObject, or is a PSObject with a base object.
+                case string:
+                case BigInteger:
+                case var _ when baseType.IsEnum || baseType.IsPrimitive:
+                    return baseObj.ToString();
```

Comment:
Please review the cases that are covered here, and let me know if any other cases should be added.

Reply:
We could take something from our TryFastTrackPrimitiveTypes() method.

Reply:
Uri value might be worth logging.

Reply:
Added `Guid`, `Uri`, `Version`, and `SemanticVersion` to the list.

@iSazonov I cannot find the `TryFastTrackPrimitiveTypes` method.

Reply:
> @iSazonov I cannot find the `TryFastTrackPrimitiveTypes` method.

Ye it's a nested method https://github.com/PowerShell/PowerShell/blob/397bd87cb30382c56ecdd1a715f05ed2855ca29e/src/System.Management.Automation/engine/MshObject.cs#L1260-L1294

Reply:
Thanks! Added `decimal` to the list.

Reply:
Also re-arrange the order of the `case` clauses, based on the likelihood of those types being used as arguments.

Reply:
Use `if` statement instead of `switch` because the `switch` statement generates IL to check "is the instance of this type" for every `case` branch that matches a type, which is unnecessary. Also, it's easier to arrange the comparison order with `if` statements.